ere-zisk doesn't compile

Discovered when trying zisk in https://github.com/eth-act/zkevm-benchmark-workload/pull/141

To reproduce in `master`:
```
$ cargo build -p ere-zisk                     
   Compiling ere-zisk v0.0.11 (/home/ignacio/code/ere/crates/ere-zisk)
error[E0277]: the trait bound `Arc<dyn erased_serde::ser::Serialize + Send + Sync>: Serialize` is not satisfied
   --> crates/ere-zisk/src/lib.rs:291:51
    |
291 |                 bincode::serialize_into(&mut acc, obj)?;
    |                 -----------------------           ^^^ the trait `Serialize` is not implemented for `Arc<dyn erased_serde::ser::Serialize + Send + Sync>`
    |                 |
    |                 required by a bound introduced by this call
    |
    = note: for local types consider adding `#[derive(serde::Serialize)]` to your `Arc<dyn erased_serde::ser::Serialize + Send + Sync>` type
    = note: for types from other crates check whether the crate offers a `serde` feature flag
    = help: the following other types implement trait `Serialize`:
              &'a T
              &'a mut T
              ()
              (T,)
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
            and 147 others
note: required by a bound in `bincode::serialize_into`
   --> /home/ignacio/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bincode-1.3.3/src/lib.rs:93:8
    |
90  | pub fn serialize_into<W, T: ?Sized>(writer: W, value: &T) -> Result<()>
    |        -------------- required by a bound in this function
...
93  |     T: serde::Serialize,
    |        ^^^^^^^^^^^^^^^^ required by this bound in `serialize_into`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ere-zisk` (lib) due to 1 previous error
```

Sounds weird this didn't happen in CI? Or mabye there's something wrong in my machine.

--- a/.github/workflows/rust-checks.yml
+++ b/.github/workflows/rust-checks.yml
@@ -27,6 +27,46 @@ jobs:
       - name: Check formatting
         run: cargo fmt --check --all
 
+  check-clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - zkvm: jolt
+            toolchain: 1.85.0
+          - zkvm: nexus
+            toolchain: nightly-2025-06-05
+          - zkvm: openvm
+            toolchain: 1.85.0
+          - zkvm: pico
+            toolchain: nightly-2024-11-27
+          - zkvm: risc0
+            toolchain: 1.85.0
+          - zkvm: sp1
+            toolchain: 1.85.0
+          - zkvm: zisk
+            toolchain: 1.85.0
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          components: clippy
+      
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
+
+      - name: Check clippy ere-${{ matrix.zkvm }}
+        run: cargo +${{ matrix.toolchain }} clippy --bins --lib --examples --tests --benches -p ere-${{ matrix.zkvm }}
+
+      - name: Check clippy ere-cli with feature ${{ matrix.zkvm }}
+        run: cargo +${{ matrix.toolchain }} clippy --bins --lib --examples --tests --benches -p ere-cli --features cli,${{ matrix.zkvm }}
+
   check-tests:
     name: Tests
     runs-on: ubuntu-latest
@@ -46,14 +86,9 @@ jobs:
       
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@nightly
-        with:
-          components: clippy
       
       - name: Cache dependencies
         uses: Swatinem/rust-cache@v2
-
-      - name: Check clippy
-        run: cargo clippy --bins --lib --examples --tests --benches -p ere-${{ matrix.zkvm }}
 
       - name: Run tests
         run: cargo test --release -p ere-dockerized -- ${{ matrix.zkvm }}

--- a/crates/ere-cli/src/lib.rs
+++ b/crates/ere-cli/src/lib.rs
@@ -1,3 +1,1 @@
-#![cfg_attr(not(test), warn(unused_crate_dependencies))]
-
 pub mod serde;

--- a/crates/ere-zisk/src/lib.rs
+++ b/crates/ere-zisk/src/lib.rs
@@ -288,7 +288,7 @@ fn serialize_inputs(inputs: &Input) -> Result<Vec<u8>, bincode::Error> {
     inputs.iter().try_fold(Vec::new(), |mut acc, item| {
         match item {
             InputItem::Object(obj) => {
-                bincode::serialize_into(&mut acc, obj)?;
+                bincode::serialize_into(&mut acc, &**obj)?;
             }
             InputItem::SerializedObject(bytes) | InputItem::Bytes(bytes) => acc.extend(bytes),
         };

